### ISSUE TITLE
Wrong GridLength Value Unit when changed by GridSplitter

### ISSUE THREAD

**Issue by guilhermeolisi** [AvaloniaUI/Avalonia#7885]
**Describe the bug**
When Width of Column or Height of a Row of a grid is Star unit and the user change the Length by a GridSplitter the value property of the GridLength object is not in Star Unit but in Pixel Unit.

**To Reproduce**
I made a example to check this issue, the attached file below. The Height of 4th Row of the grid is binding and converted to a double property  of the ViewModel. The initial value of this property is 2* (-2) and it is correct when the view is loaded. If you first click in button "Try to set Length to 1*" it will work correctly. However, if you change the Height by GridSplitter the value of double property in VM is in Pixel Unit and no longer in Star Unit. If after changing the Height by GridSplitter you click in the button again the Height of the Row will be wrong.
[TestGridLength.zip](https://github.com/AvaloniaUI/Avalonia/files/8358640/TestGridLength.zip)

**Expected behavior**
The double property in VM must be in Star Unit, not in Pixel Unit.

**Desktop (please complete the following information):**

- OS: Windows 10
- Version (avalonia): 0.10.12 and above

**Comment by guilhermeolisi** [comment]
Hi, thanks for your reply.
I understand that the changes are in absolute size, but I think it would be very useful if the Value property of GridLength followed the Unit defined for this object. Switching to the Pixel unit at some point without me explicitly changing the Unit causes a lot of confusion. 
The behavior of this property in WPF is as I describe my expectation, when I change the GridLength by GridSplitter, the returned value follows the unit defined for the object.

**Comment by robloo** [comment]
> The behavior of this property in WPF is as I describe my expectation

Well, if that's what WPF did Avalonia should as well. So I'll agree with you overall. I think this is a feature request more than a bug though.

**Comment by guilhermeolisi** [comment]
Hi @robloo,
Do you think it is necessary for me to open a "New Issue" as "Feature Request" instead of "Bug report" for this modification?

**Comment by robloo** [comment]
> Do you think it is necessary for me to open a "New Issue" as "Feature Request" instead of "Bug report" for this modification?

No, I don't think it's necessary at all. It's good to keep this discussion all in one place as well.

One of the maintainers might re-label it in the future. It's not a problem either way.

**Comment by guilhermeolisi** [comment]
Ok, thanks
